### ISSUE TITLE
Uploaded logo does not have alt text

### ISSUE THREAD

**Issue by adriannelee** [cds-snc/notification-planning#557]
![image.png](https://images.zenhubusercontent.com/61267d15419dc0a24aa009dd/45c659bd-0607-4a4c-afa6-040eba60e269)
## User story
As a Sender, when I'm requesting a new email branding using a custom logo, I am required to also include alt text so that my email is accessible.

## Why are we building
A11y issue - On the Settings page, you are presented the options to upload a logo for the service. The uploaded logo is displayed on the “Your logo is being reviewed” page, however it does not include any alt text. Ideally the user should be allowed to specify the appropriate alt text when uploading.

Issue group: Settings
Visibility: 1
Evaluation: Partially Supports
Success Criteria: 1.1.1: Non-text Content
Needs design: Interaction, Content

## What are we building
When Senders upload a custom logo on the "Change your email branding" flow, Notify provides them with auto-generated alt-text using their service name. The Sender can modify the alt-text. The field is required. The auto-generated text is bilingual, according to which language they've set (English-first/French-first). Some guidance is provided in case they want to modify the auto-generated option.

## Value
- A11y by default
- Make it easy for Senders to create good alt text

## Considerations
- An alternative design could be where the Sender chooses whether they want to auto-generate or if they want to write their own
- For auto-generating, in French it could be "Logo de .. ", "Logo du .. ", "Logo d'"
- Alt text guidance in English is to say the name of the org and "logo" at the end of the alt text (e.g. "Canadian Digital Service logo")
- We want to keep this flow modular (not embed it into the 'create a service' flow or the 'go live request' flow
- Notify team should take on the load manually to update Alt text for legacy services and one offs moving forward until we have the self serve option, so Support doesn't need to take this extra work on
- Notify team could draft alt text for each custom logo and seek approval of the service, before uploading it ourselves, once this feature is built. 
- Notify could provide templates for logos so services just need to fill in the blanks
 
## Links
[Figma](https://www.figma.com/file/a1TQZWGPRijQkAfa7DHWXF/Service-branding-%7C-Image-de-service?type=design&node-id=3-109&t=Gi3rZSnYWYfje1di-4)
[2 releases outlined](https://miro.com/app/board/uXjVO6TPyIs=/?moveToWidget=3458764561800700364&cot=14)
[scoping notes Jumana/Yael](https://docs.google.com/document/d/1Rtrz3ehmlHJ2b-apuVf4AwZBD-p1puwjxXrsQMmmNAk/edit)

## Acceptance Criteria

- [ ] Find out how many services currently have a custom logo?
- [ ] When a recipient is reading the email using alt text, the alt text is there.
- [ ] The guidance to write good alt text for a logo is provided to Senders when they're completing the form
- [ ] Autogeneration of alt-text works reasonably well for most scenarios, in both languages
- [ ] The autogenerated text is clear that it is auto-generated and not there by mistake

**Comment by yaelberger-commits** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Notify-Planning-614b3ad91bc2030015ed22f5/issues/cds-snc/notification-planning/557?planning-poker) with Zenhub @amazingphilippe @whabanks

**Comment by anikbrazeau** [comment]
Question: Does Notify collect the organization of all users/account/services? I wonder if maybe this could be the alt text by default in EN/FR? Or are there potentially cases where the department/organization would not match the logo?

**Comment by whabanks** [comment]
@amazingphilippe As we touched briefly on it in the dev <-> design sync, I did some digging into the widths for logos from our email templates and this is what I found:

[Base email template](https://github.com/cds-snc/notification-utils/blob/main/notifications_utils/jinja_templates/email/email_template.jinja2):
max-width: 580px

[Custom logo with a background colour](https://github.com/cds-snc/notification-utils/blob/main/notifications_utils/jinja_templates/email/_custom_logo_with_background_colour.jinja2):
max-width: 580px
padding-left & padding-right: 10px
Available width: 560px

[Custom logo without a background colour](https://github.com/cds-snc/notification-utils/blob/main/notifications_utils/jinja_templates/email/_custom_logo_no_background_colour.jinja2)
max-width: 580px
Padding-right: 10px
padding-left: 8px (if there is brand color) 10px otherwise
Available width: 560px or 562px

580px seems like a deliberate choice for a width. Both [Mailchimp](https://templates.mailchimp.com/getting-started/html-email-basics/#:~:text=Best%20Practices%3A%20Design&text=Emails%20should%20be%20600%2D800,require%20HTML%20floats%20or%20positioning.) and [Taxi for email](https://taxiforemail.com/blog/whats-the-optimal-width-for-your-email-design/) recommend anywhere from 600-800px as a max-width. Should 580px become a limiting factor perhaps we can revisit and discuss the rationale for going with 580px in the future.

**Comment by YedidaZalik** [comment]
Included in Ally statement

**Comment by yaelberger-commits** [comment]
Yael & Jumana to meet and do some scoping and planning for this feature April 4

**Comment by yaelberger-commits** [comment]
Ready for Story Refinement

**Comment by andrewleith** [comment]
- @jzbahrai and @yaelberger-commits have discussed and posted ideas in our branding channel

**Comment by YedidaZalik** [comment]
Will discuss today at story refinement
Hopefully will work on next sprint

**Comment by foudamo** [comment]
@jzbahrai turn this into an epic, and create cards based on the document

**Comment by YedidaZalik** [comment]
Jumana working on this and picked up a sub-ticket
